plugins/flytekit-bigquery/flytekitplugins/bigquery/agent.py

```diff
@@ -73,54 +63,36 @@ def create(
 
-        return CreateTaskResponse(resource_meta=json.dumps(asdict(metadata)).encode("utf-8"))
+        return BigQueryMetadata(job_id=str(query_job.job_id), location=location, project=project)
 
-    def get(self, resource_meta: bytes, **kwargs) -> GetTaskResponse:
+    def get(self, metadata: BigQueryMetadata, **kwargs) -> Resource:
         client = bigquery.Client()
-        metadata = Metadata(**json.loads(resource_meta.decode("utf-8")))
-        log_links = [
-            TaskLog(
-                uri=f"https://console.cloud.google.com/bigquery?project={metadata.project}&j=bq:{metadata.location}:{metadata.job_id}&page=queryresults",
-                name="BigQuery Console",
-            ).to_flyte_idl()
-        ]
+        log_link = TaskLog(
+            uri=f"https://console.cloud.google.com/bigquery?project={metadata.project}&j=bq:{metadata.location}:{metadata.job_id}&page=queryresults",
+            name="BigQuery Console",
+        )
 
         job = client.get_job(metadata.job_id, metadata.project, metadata.location)
         if job.errors:
             logger.error("failed to run BigQuery job with error:", job.errors.__str__())
-            return GetTaskResponse(
-                resource=Resource(state=TaskExecution.FAILED, message=job.errors.__str__()), log_links=log_links
-            )
+            return Resource(phase=TaskExecution.FAILED, message=job.errors.__str__(), log_links=[log_link])
 
         cur_phase = convert_to_flyte_phase(str(job.state))
         res = None
 
         if cur_phase == TaskExecution.SUCCEEDED:
-            ctx = FlyteContextManager.current_context()
-            if job.destination:
-                output_location = (
-                    f"bq://{job.destination.project}:{job.destination.dataset_id}.{job.destination.table_id}"
-                )
-                res = literals.LiteralMap(
-                    {
-                        "results": TypeEngine.to_literal(
-                            ctx,
-                            StructuredDataset(uri=output_location),
-                            StructuredDataset,
-                            LiteralType(structured_dataset_type=StructuredDatasetType(format="")),
-                        )
-                    }
-                ).to_flyte_idl()
-
-        return GetTaskResponse(resource=Resource(phase=cur_phase, outputs=res), log_links=log_links)
-
-    def delete(self, resource_meta: bytes, **kwargs) -> DeleteTaskResponse:
+            dst = job.destination
+            if dst:
+                ctx = FlyteContextManager.current_context()
+                output_location = f"bq://{dst.project}:{dst.dataset_id}.{dst.table_id}"
+                res = TypeEngine.dict_to_literal_map(ctx, {"results": StructuredDataset(uri=output_location)})
+
+        return Resource(phase=cur_phase, message=job.state, log_links=[log_link], outputs=res)
+
+    def delete(self, metadata: BigQueryMetadata, **kwargs):
```

Comment:
`metadata` needs to be replaced with `resource_meta`

Reply:
yes, good catch, thanks. I just updated it.